app/presenters/notification_presenter.rb

```diff
@@ -19,6 +19,24 @@ def list(raw_notifications)
       end
     end
 
+    def list_by_cluster_id(raw_notifications, cluster_id)
+      notifications = []
+      raw_notifications.each do |notification|
+        notification.each do |notification_id, attributes|
+          next unless attributes['cluster_id'] == cluster_id
```

Comment:
Dont think there is a field named `cluster_id` in output of `/notifications` REST endpoint. A sample looks as below

```
[{
  "message_id": "75e2c539-1265-443a-8979-b662a806dcb2",
  "timestamp": "2018-01-31T12:37:33.672914+00:00",
  "priority": "notice",
  "message": "Cluster gluster-373468a41b2f2405951e42e64d8df9eecf31d5b1956496e90a0a41261f34418c moved to unhealthy state"
}, {
  "message_id": "24858a68-f367-4831-8ca3-ac07b30a8f8e",
  "timestamp": "2018-01-31T12:36:05.373378+00:00",
  "priority": "notice",
  "message": "Node dhcp41-247.lab.eng.blr.redhat.com is UP"
}, {
  "message_id": "552d9043-8450-434f-86de-cf4aa345a848",
  "timestamp": "2018-01-31T12:36:01.117541+00:00",
  "priority": "notice",
  "message": "Node dhcp43-19.lab.eng.blr.redhat.com is UP"
}, {
  "message_id": "a466e9d7-496b-4ab9-b8f2-e91ca060cd9e",
  "timestamp": "2018-01-31T12:35:51.370553+00:00",
  "priority": "notice",
  "message": "Node dhcp43-29.lab.eng.blr.redhat.com is UP"
}, {
  "message_id": "e4d17304-c3e7-4cc7-95bf-3ad849444f0b",
  "timestamp": "2018-01-31T12:31:37.590025+00:00",
  "priority": "notice",
  "message": "Node dhcp43-24.lab.eng.blr.redhat.com is UP"
}, {
  "message_id": "9daf583b-daa9-49d5-b5d1-84dfe14a0b33",
  "timestamp": "2018-01-31T12:30:26.531524+00:00",
  "priority": "notice",
  "message": "Node dhcp43-29.lab.eng.blr.redhat.com is UP"
}]
```

Reply:
Why do you need cluster_id in response? Check L27 where the attributes are selected. Before that we are simply checking if cluster_id matches.